Dockerfile

```diff
@@ -22,38 +22,33 @@
 
 # ------------------------------------------------------------------------------
 # Cargo Build Stage
-#
-# Runs on native host architecture
-# Cross compiles for target architecture
 # ------------------------------------------------------------------------------
-FROM --platform=$BUILDPLATFORM rust:latest AS cargo-build
-RUN apt-get update && \
-    apt-get upgrade -y && \
-    apt-get install -y cmake musl-tools clang llvm protobuf-compiler
+FROM rust:alpine3.17 AS cargo-build
+ARG FEATURES
+RUN apk add --no-cache --update \
+    clang15-libclang \
+    cmake \
+    g++ \
+    gcc \
+    libc-dev \
+    musl-dev \
+    protobuf \
+    tpm2-tss-dev
 
 WORKDIR /tmp/helium_gateway
 COPY . .
 
-ENV CC_aarch64_unknown_linux_musl=clang
-ENV AR_aarch64_unknown_linux_musl=llvm-ar
-ENV CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_RUSTFLAGS="-Clink-self-contained=yes -Clinker=rust-lld"
-
-ENV CC_x86_64_unknown_linux_musl=clang
-ENV AR_x86_64_unknown_linux_musl=llvm-ar
-ENV CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_RUSTFLAGS="-Clink-self-contained=yes -Clinker=rust-lld"
+ENV CC=gcc CXX=g++ CFLAGS="-U__sun__" RUSTFLAGS="-C target-feature=-crt-static"
 
-ARG TARGETPLATFORM
-RUN \
-case "$TARGETPLATFORM" in \
-    "linux/arm64") echo aarch64-unknown-linux-musl > rust_target.txt ;; \
-    "linux/amd64") echo x86_64-unknown-linux-musl > rust_target.txt ;; \
-    *) exit 1 ;; \
-esac
+# TMP build fail when cross compiling, so we need to use QEMU when
+# building for not-host architectures. But QUEMU builds fail in CI due
+# to OOMing on cargo registry updating. Therefore, we will need to
+# compile with nightly until cargo's sparse registry stabilizes.
```

Comment:
i believe i heard this was supposed to land in the 1.68 release

Reply:
Cool. Let’s try to remember to revisit this when it does